src/lib/directories/configFile.ts

```diff
@@ -0,0 +1,56 @@
+import { Dialog, DialogChainObject } from "quasar"
+import * as dialog from "@tauri-apps/api/dialog"
+import * as fs from "@tauri-apps/api/fs"
+import * as path from "@tauri-apps/api/path"
+import { invoke } from "@tauri-apps/api/tauri"
+import * as process from "process"
+import { ConfigFile, emptyConfig } from "../util"
+
+const tauri = { dialog, fs, path, invoke }
+
+export const dirName = process.env.DEFAULT_APP_DIR || "subspace-desktop"
+
+/**  
+  Linux: Resolves to $XDG_CONFIG_HOME or $HOME/.config.
+  macOS: Resolves to $HOME/Library/Application Support.
+  Windows: Resolves to {FOLDERID_LocalAppData}.
+ */
+export const configFile = {
+  async getConfigFile(): Promise<ConfigFile | void> {
+    const configDir = (await tauri.path.configDir()) + dirName
+    const result = await tauri.fs
+      .readTextFile(configDir + "/config.json")
+      .catch(console.error)
+    if (result) return JSON.parse(result)
+  },
+  async updateConfigFile(newData: ConfigFile): Promise<void> {
+    const configFile = (await this.getConfigFile()) || emptyConfig
+    for (const [key, value] of Object.entries(newData)) {
+      Object.assign(configFile[key], value)
+    }
+    await this.writeConfigFile(configFile)
+  },
+  async writeConfigFile(config: ConfigFile): Promise<void> {
+    const configDir = (await tauri.path.configDir()) + dirName
+    await tauri.fs
+      .writeFile({
+        path: configDir + "/config.json",
```

Comment:
Are you sure you need a config file? Local Storage seems to be persistent, maybe use that instead? Or these are for fundamentally different purpose? (I already saw some of the similar data stored in Local Storage and config file (like path to config is in Local Storage).

Reply:
> path to config is in Local Storage.

- Should not be in this branch. It is not required anymore as we now use `OS config directory`, the app always knows where the **config.json file lives**.

> Are you sure you need a config file? Local Storage seems to be persistent, maybe use that instead?

- 👍 agree with replacing the use of `config.json` and using **LocalStorage** instead.
Will remove a **lot of code**, **few files**, will **reduce code** on pages working on this file.
And finally as an extra, will **fix the macOS problem** on using the same OS directory for **config** and default application **data**.

> This config.json came from the initial code, the demo version John develop. I initially assumed that it was necessary for **tauri or quasar**, but after working in the app, removing data from it, updating the structure also I have realized that NOW is more a mini **util/cache** than a **config** itself.

- Do we move it to LocalStorage? I say yes and with a big smile: One less folder to handle, less invasive to the filesystem, and actually, will fix this problem on macOS using same directory for config and app data.